### PR TITLE
Pool

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have realized design pattern named Pool.

### CODE REVIEW

File: src/com/clouway/pool/PooledObject.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.clouway.pool;
+
+import java.util.Random;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class PooledObject {
```
Comment (freekman) [inline src/com/clouway/pool/PooledObject.java]: rename to IP

File: src/com/clouway/pool/ObjectPool.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.clouway.pool;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class ObjectPool {
```
Comment (freekman) [inline src/com/clouway/pool/ObjectPool.java]: rename to IpPool

File: src/com/clouway/pool/Demo.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public static void main(String[] args) throws Exception {
+        IpPool pool = new IpPool(3);
+
+        IP IP1 = pool.acquire();
+        IP IP2 = pool.acquire();
+        IP IP3 = pool.acquire();
+
+        System.out.println("Acquire " + IP1);
+        System.out.println("Acquire " + IP2);
+        System.out.println("Acquire " + IP3);
```
Comment (freekman) [inline src/com/clouway/pool/Demo.java]: add the case when the pool is empty and you make an acquire
you could have added tests

Comment (Stanislava88) [inline src/com/clouway/pool/Demo.java]: done

**Comment by Stanislava88** [comment]
done

File: src/com/clouway/pool/NoFreeResourceException.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.clouway.pool;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class NoFreeResourceException extends RuntimeException {
+    public NoFreeResourceException(String msg) {
```
Comment (freekman) [inline src/com/clouway/pool/NoFreeResourceException.java]: leave it only 
public class NoFreeResourceException extends RuntimeException {}

File: src/com/clouway/pool/Demo.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package com.clouway.pool;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class Demo {
+    public static void main(String[] args) throws Exception {
```
Comment (freekman) [inline src/com/clouway/pool/Demo.java]: with the tets added you could remove the main method

File: src/com/clouway/pool/IpPool.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class IpPool {
+    private Queue<IP> ips;
+    private Integer maxNum;
```
Comment (freekman) [inline src/com/clouway/pool/IpPool.java]: final

**Comment by freekman** [comment]
LGTM after fix comments

File: src/com/clouway/pool/IpPool.java (hunk 6 above)
Comment (Stanislava88) [inline src/com/clouway/pool/IpPool.java]: done

File: src/com/clouway/pool/Demo.java (hunk 5 above)
Comment (Stanislava88) [inline src/com/clouway/pool/Demo.java]: done

File: src/com/clouway/pool/NoFreeResourceException.java (hunk 4 above)
Comment (Stanislava88) [inline src/com/clouway/pool/NoFreeResourceException.java]: done

**Pull request merged**